flow/packet-headers/ipv4.yaml

```diff
@@ -122,6 +122,100 @@ components:
             default: 0.0.0.0
             features: [count, metric_tags]
           x-field-uid: 14
+        options:
+          type: array
+          minItems: 0
+          items:
+            $ref: '#/components/schemas/Flow.Ipv4.Options'
+          x-field-uid: 15
+    Flow.Ipv4.Options:
+      description: >-
+        IPv4 options are optional extensions for the IPv4 header that can be utilised to provide additional information about the IPv4 datagram. 
+        It is encoded as a series of type, length and value attributes. 
+        The IP header length MUST be increased to accommodate the extra bytes needed to encode the IP options.
+        Currently IP options supported are:
+        1. router_alert Option allows devices to intercept packets not addressed to them directly as defined in RFC2113.
+        2. custom option is provided for to be able to configure user defined IP options as needed.
+      type: object
+      properties:
+        choice:
+          type: string
+          default: router_alert
+          x-field-uid: 1
+          x-enum:
+            router_alert:
+              x-field-uid: 1
+            custom:
+              x-field-uid: 2
+        custom:
+          $ref: '#/components/schemas/Flow.Ipv4.Custom'
+          x-field-uid: 3
+    Flow.Ipv4.Custom:
+      description: >-
+        User defined IP options to be appended to the IPv4 header.
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/Flow.Ipv4.Type'
+          x-field-uid: 1
+        length:
+          x-field-pattern:
+            description: >-
+              Length of the option field
+            format: integer
+            length: 8
+            default: 4
+            features: [count]
+          x-field-uid: 2
+        value:
+          x-field-pattern:
+            description: >-
+              Length of the option field
+            format: integer
```

Comment:
Since this is a custom option, the value field should probably be a string restricted to take input of a hex string to truly allow custom options to be added. We will either need to throw error/warning  in implementation if hex bytes length do not match with length and auto update both options length and IP header length when creating packet OR just create the packet as provided and specify explicitly that for custom option , IP header length needs to be set explicitly by user to create valid packet .  Probably 1st is more user-friendly but second allows for creation of back IP packets with incorrect length etc. with IP options + IP options with bad length etc. 
Probably providing auto option for length with value override might handle both .